stan/math/rev/mat/fun/Eigen_NumTraits.hpp

```diff
@@ -79,6 +79,10 @@ struct NumTraits<stan::math::var> : GenericNumTraits<stan::math::var> {
 };
 
 namespace internal {
+template <>
+struct remove_all<stan::math::vari*> {
+  typedef stan::math::vari* type;
+};
```

Comment:
Not seeing this used anywhere?

Reply:
Forgot to add doc for this. Needed to add this specialisation to stop Eigen from removing pointers when working with ```vari*```. Otherwise the ```.vi()``` method would return ```Matrix<vari,R,C>``` rather than ```Matrix<vari*,R,C>```